pkg/operator/ceph/client/controller.go

```diff
@@ -227,28 +235,55 @@ func (r *ReconcileCephClient) reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, *cephClient, errors.Wrapf(err, "failed to validate client %q arguments", cephClient.Name)
 	}
 
+	// Detect desired CephCluster version
+	runningCephVersion, desiredCephVersion, err := currentAndDesiredCephVersion(
+		r.opManagerContext,
+		r.opConfig.Image,
+		cephClient.Namespace,
+		controllerName,
+		k8sutil.NewOwnerInfo(cephClient, r.scheme),
+		r.context,
+		r.clusterSpec,
+		r.clusterInfo,
+	)
+	if err != nil {
+		if strings.Contains(err.Error(), opcontroller.UninitializedCephConfigError) {
+			logger.Info(opcontroller.OperatorNotInitializedMessage)
+			return opcontroller.WaitForRequeueIfOperatorNotInitialized, *cephClient, nil
+		}
+		return reconcile.Result{}, *cephClient, errors.Wrap(err, "failed to detect running and desired ceph version")
+	}
```

Comment:
As discussed in huddle, since we aren't implementing `WithCephVersionUpdate`, this isn't necessary yet. I think we can replace this with `cephclient.LeastUptodateDaemonVersion()`, using the output for both running and desired to `ShouldRotateCephxKeys()`.

Add a comment to note that this will need adjusted when/if `WithCephVersionUpdate` is implemented. This may also be a matter of if it is implemented. CephClient doesn't benefit from `WithCephVersionUpdate` as much as CephCluster, CephFilesystem, and CephObjectStore, so we may choose not to implement it for CephClient unless requested by users.